### PR TITLE
show recurring only when applicable

### PR DESCRIPTION
**Pull request by kalidiagne**
# Description

**Lock with recurring applicable**

Screenshot 2023-04-05 at 12 18 56

**Lock without recurring applicable**

Screenshot 2023-04-05 at 12 10 41

# Issues

Fixes #
Refs #

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

**Review by julien51 — COMMENTED** [review]
Did you check that the checkout still works when recurring is set to 0? 
I suspect this may have broken things.

File: unlock-app/src/components/interface/locks/CheckoutUrl/elements/LocksForm.tsx [hunk 1]
```diff
@@ -90,11 +98,39 @@ export const LocksForm = ({
...
   const [recurringUnlimited, setRecurringUnlimited] = useState(false)
+  const [lockRecurring, setLockRecurring] = useState<RecurringByLock>({})
 
   const { getIsRecurringPossible } = useLockSettings()
 
   const [locks, setLocks] = useState<LocksProps>(locksDefault)
 
+  // preload default and set recurring (es. saved config)
+  useEffect(() => {
+    const getRecurringCb = async () => {
```
Comment (julien51) [inline unlock-app/src/components/interface/locks/CheckoutUrl/elements/LocksForm.tsx:109]: Please use a better function name

Comment (julien51) [reply unlock-app/src/components/interface/locks/CheckoutUrl/elements/LocksForm.tsx:109]: Also I think I have already seen you write code to check if a lock is recurring somewhere? Are you sure your code is not duplicated?

Comment (kalidiagne) [reply unlock-app/src/components/interface/locks/CheckoutUrl/elements/LocksForm.tsx:109]: We need both: 
- when we add a new lock to the list, check if recurring is possible 
- for an existing saved config with locks, we fetch with the code above

**Comment by kalidiagne** [comment]
> Did you check that the checkout still works when recurring is set to 0? I suspect this may have broken things.

Yes, there is no issue on this side.
The only is on Metamask side with the `default value` on approve: without setting the default value but a custom value it fails :/ (also when renewals > 0)

**Review by julien51 — APPROVED** [review]

**Review by julien51 — APPROVED** [review]

**Pull request merged**